types/redux-first-router/index.d.ts

```diff
@@ -179,9 +194,9 @@ export interface Options {
     location?: string | SelectLocationState;
     notFoundPath?: string;
     scrollTop?: boolean;
-    onBeforeChange?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
-    onAfterChange?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
-    onBackNext?<S>(dispatch: Dispatch<S>, getState: StateGetter): void;
+    onBeforeChange?(dispatch: Dispatch<any>, getState: StateGetter): void;
```

Comment:
same as above - wouldn't explicitly typing the shape of the store be preferred?

Reply:
Based on the [`no-unnecessary-generics`](https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md) dtslint rule, it's not needed because the generic type was only used once.